grakn-graql/src/main/java/ai/grakn/graql/internal/query/ConceptBuilder.java

```diff
@@ -90,14 +90,38 @@
     private final Map<BuilderParam<?>, Object> params = new HashMap<>();
```

Comment:
This would also mean `expect(BuilderParam)` would need to turn into `used(BuilderParam)` which i think is a good change anyway.